ambry-api/src/main/java/com.github.ambry/rest/RequestPath.java

```diff
@@ -160,10 +188,17 @@ public String getOperationOrBlobId(boolean stripLeadingSlash) {
   /**
    * @return a {@link RestUtils.SubResource}, or {@code null} if no sub-resource was found in the request path.
    */
-  public RestUtils.SubResource getSubResource() {
+  public SubResource getSubResource() {
     return subResource;
   }
 
+  /**
+   * @return blob segment number for segmented blobs, or NO_BLOB_SEGMENT_IDX_SPECIFIED for non-segmented blobs
```

Comment:
minor: update this comment `blob segment index`

Reply:
done